mlflow/projects/__init__.py

```diff
@@ -37,24 +37,25 @@ def _run(uri, entry_point="main", version=None, parameters=None, experiment_id=N
     """
     exp_id = experiment_id or _get_experiment_id()
     parameters = parameters or {}
+    work_dir = _fetch_project(uri=uri, force_tempdir=False, version=version,
```

Comment:
The project-fetching behavior is slightly different for databricks runs  - we always fetch into a temp directory & then remove the `mlruns` directory (if one exists) to prevent cache-busting when uploading the tarred project to DBFS. Maybe we can continue to allow the individual `mode`s to handle project-fetching? Other than that, the change of calling `projects._create_run` in both modes looks good.

Reply:
Oh yeah, thats'a a good catch. But why don't we do the same for local git projects?